test/Npgsql.Tests/CommandTests.cs

```diff
@@ -348,6 +348,366 @@ public void CloseConnectionWithException()
+                        cmd.Parameters["p0"].Value = "test2";
+                        using (var dr = cmd.ExecuteReader())
+                        {
+                            Assert.IsNotNull(dr);
+                            dr.Close();
+                            Assert.That(dr.RecordsAffected, Is.EqualTo(1));
+                        }
+                        Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1));
+                    }
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM data WHERE name = 'test2'"), Is.EqualTo(1));
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1), "Persistent prepared statement deallocated");
+
+                    Assert.That(conn.ExecuteScalar("SELECT name FROM pg_prepared_statements LIMIT 1") as string, Is.EqualTo(stmtName), "Persistent prepared statement name changed unexpectedly");
+                }
+            }
+            finally
+            {
+                // Clear the pools so that we dont mess up other tests that depend
+                // on the clean state of pg_prepared_statements
+                NpgsqlConnection.ClearAllPools();
+            }
+        }
+
+        [Test, Description("Basic persistent prepared system scenario. Checks that statement is not deallocated in the backend after connection close.")]
+        public void PersistentPrepareAfterConnectionClose()
+        {
+            try
+            {
+                string stmtName;
+
+                var conn = OpenConnection();
+                conn.ExecuteNonQuery("CREATE TEMP TABLE data (name TEXT)");
+                Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(0));
+
+                using (var cmd = new NpgsqlCommand("INSERT INTO data (name) VALUES (:p0);", conn))
+                {
+                    cmd.IsPersistent = true;
+
+                    cmd.Parameters.Add(new NpgsqlParameter("p0", NpgsqlDbType.Text));
+                    cmd.Prepare();
+                    cmd.Parameters["p0"].Value = "test1";
+                    using (var dr = cmd.ExecuteReader())
+                    {
+                        Assert.IsNotNull(dr);
+                        dr.Close();
+                        Assert.That(dr.RecordsAffected, Is.EqualTo(1));
+                    }
+                    Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1));
+                }
+                Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM data WHERE name = 'test1'"), Is.EqualTo(1));
+                Assert.That(conn.ExecuteScalar("SELECT COUNT(*) FROM pg_prepared_statements"), Is.EqualTo(1), "Persistent prepared statement deallocated");
+
+                stmtName = conn.ExecuteScalar("SELECT name FROM pg_prepared_statements LIMIT 1") as string;
+                conn.Close();
+
+                conn.Open();
```

Comment:
This needs a check that we got the same pooled connection by comparing the new connection's ProcessId with the old's.